### DISCUSSION TITLE
What permissions required to get repository environments in API v4 for github app?

### DISCUSSION THREAD

**Discussion by L1ghtman2k** [community/community#55354] [Apps, API and Webhooks]
### Select Topic Area

Question

### Body

I have the following GraphQL query that I am able to run using my PAT:
```
query EnvironmentsView($owner: String!, $name: String!, $first: Int!, $cursor: String) {
    repository(owner: $owner, name: $name) {
        environments(first: $first, after: $cursor) {
            pageInfo {
                endCursor
                hasNextPage
            }
            totalCount
            edges {
                cursor
                node {
                    name
                }
            }
        }
    }
}
```

Whenever running above query with GithubApp Token (after requesting installation token), I get this error:
```
(
    [
        Error {
            message: "Resource not accessible by integration",
            locations: Some(
                [
                    Location {
                        line: 7,
                        column: 9,
                    },
                ],
            ),
            path: Some(
                [
                    Key(
                        "repository",
                    ),
                    Key(
                        "environments",
                    ),
                ],
            ),
            extensions: Some(
                {
                    "saml_failure": Bool(false),
                },
            ),
        },
    ],
)
```

My GitHub app had the following repository permissions at the time of running the query:

- Administration: Read & write
- Contents: Read & write
- Metadata: Read
- Environments: Read
- Checks: Read & write (probably irrelevant)

What permissions should I use to query repository environments?

**Comment by jge162** [comment]
Hello @L1ghtman2k please update `Environments: Read & write` and rerun. Bests

**Reply by L1ghtman2k** [reply]
@jge162 , I have done that too, and that did not help.

I have in fact tried giving all write permissions repo permissions to the app, and that didn't work

**Reply by jge162** [reply]
In addition, ensure that the variables $owner and $name in your GraphQL query are correctly set to the repository owner and name, respectively.

**Reply by L1ghtman2k** [reply]
Correct, I just confirmed it. I have tried running the same query with different credentials. My PAT works, but my Github APP Installation token does not. I could tell my GithubAPP installation token works, because I can list branch protection rules via API, just not the environments

**Reply by jge162** [reply]
This is what I would do to try and resolve since updating environment permissions. 

- Verify App Permissions
- Reinstall the App
- Check Repository Access
- Generate a New Installation Token

**Reply by L1ghtman2k** [reply]
At this point, I have done the above multiple times. I think this is a bug with the permissions, or that this API is just not exposed to GitHub Apps, but thanks for taking your time

**Reply by jge162** [reply]
You should reach out to support or submit [bug](https://support.github.com/contact/bug-report) tbh.

**Reply by L1ghtman2k** [reply]
Oh, I was under the impression this was the place to do it, thanks a lot @jge162 !

**Comment by L1ghtman2k** [comment]
For anyone who stumbles on this in the future, you need: `Actions: Read` permissions to read environments
